go/vt/sqlparser/analyzer.go

```diff
@@ -120,13 +120,50 @@ func ASTToStatementType(stmt Statement) StatementType {
 
 //CanNormalize takes Statement and returns if the statement can be normalized.
 func CanNormalize(stmt Statement) bool {
-	switch stmt.(type) {
-	case *Select, *Union, *Insert, *Update, *Delete, *Set, *CallProc, *Stream: // TODO: we could merge this logic into ASTrewriter
+	switch s := stmt.(type) {
+	case *Select:
+		// Skip the system table normalization, normalization will cause schemaname replacement chaos.
+		return !containSystemTable(s.From)
+	case *Union:
+		// Skip union that involves system table queries.
+		if !CanNormalize(s.FirstStatement) {
```

Comment:
isn't first statement covered in union selects?

Reply:
FirstStatement is not overlapped with UnionSelects. See [Unionize](https://github.com/BlitzScaling/vitess/blob/78f0a2a3b394909de7f5b528b287941316e470f1/go/vt/sqlparser/ast_funcs.go#L836)